docs/debugging-javascript.md

```diff
@@ -0,0 +1,382 @@
+    4. Make sure the new config is selected at the top of the *Run and Debug* sidebar
+    5. Click on the ▶️ button or press `F5` in VS Code
+    > Watch VS Code's *Debug Console* output for a successful debugger connection. This can be very slow, but VS Code should automatically launch Metro, establish the debugger connection, then launch your native app to download the bundle.
+6. **Option B:** Attach VS Code to Metro that's already running
+    1. Click on the ⚙️ to the right of *Attach to packager*
+    2. A new configuration should appear in your `launch.json` file. Rename it if you want but it should look like:
+    ```json
+    {
+        "name": "Attach to packager",
+        "cwd": "${workspaceFolder}",
+        "type": "reactnative",
+        "request": "attach",
+        "isDynamic": true
+    }
+    ```
+    3. Make sure that only Metro is already running (i.e. `npx react-native start`)
+    4. Make sure the new config is selected at the top of the *Run and Debug* sidebar
+    5. Click on the ▶️ button or press `F5` in VS Code
+    6. Confirm the address of the Metro server (default: `localhost`)
+    7. Confirm the port of the Metro server (default: `8081`)
+    > Watch VS Code's *Debug Console* output for a successful debugger connection.
+
+    8. Manually launch your native app
+
+Once everything is running, you should be able to debug your RNW application. To set breakpoints, simply do so in your app's JS source files directly in VS Code.
+
+> Remember to save `launch.json` so you can reuse this debugging configuration in the future.
+
+> *Important:* If you're having difficulty getting VS Code to connect with Web Debugging, double check you're not running any part of RNW (watch out for rogue `node.exe` processes) before starting the steps above.
+
+<!--END_DOCUSAURUS_CODE_TABS-->
+
+## Direct Debugging
+
+*Direct Debugging* is the newer JS debugging solution for RN.
+
+Rather than running your app's JS code on an external JS engine (as with Web Debugging), with Direct Debugging you run your app's code on its embedded engine (as normal). You are then able to attach your debugger directly to the embedded engine.
+
+> Direct Debugging requires the Metro Packager, as Metro provides keys parts of the debugging experience such as JS source mapping.
+
+### Direct Debugging Tool Support
+
+| JavaScript Engine | Visual Studio | Visual Studio Code | Visual Studio Code<br/> w/ React Native Tools |
+|:------------------|:-:|:-:|:-:|
+| Chakra (Default)  | ✅ | 🟥 | 🟥 |
+| Hermes            | 🟥 | ✅ | ✅ |
+
+> **Important:** Direct Debugging is relatively new and may still have some rough edges, depending on your choice of engine and debugger. See [Web vs. Direct Debugging](#web-vs-direct-debugging) for details.
+
+### Step 1: Enable Direct Debugging
+
+You have two options to enable Direct Debugging: at compile-time in your app's native code or at runtime via the in-app Developer Menu.

```

Comment:
Also from CLI, yeah? `npx react-native run-windows --direct-debugging`
https://github.com/microsoft/react-native-windows/commit/c1a2140928b37d9c761c41bfa0fcc38704a14099

Reply:
Yes but I had trouble getting it to work - I think there may be a bug where those settings get overwritten.